util/test_python.bash

```diff
@@ -44,5 +44,10 @@ cd $REPO_ROOT
 log_info "Running py.test over $TST_DIR and doc examples"
 py.test --verbose --junitxml=$REPO_ROOT/python-results.xml $TST_DIR docs/source/examples/
 
+passedPytest=$?
+
 log_info "Running pych --testing..."
 pych --testing
+
+passedPych=$?
+exit $(( $passedPytest || $passedPych ))
```

Comment:
Instead of tracking failures and continuing, for travis it is often easier to add `set -e` at the top of the script. Then if anything fails (even something trivial like cloning the repo), travis will report a failure.